src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,28 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.Comparator;
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] yearLiveInUkraine = candidate.getPeriodsInUkr().split("-");
+        int periodsInUkr = Integer.parseInt(yearLiveInUkraine[1])
+                - Integer.parseInt(yearLiveInUkraine[0]);
+        if (candidate.getAge() >= 35 && candidate.isAllowedToVote()
+                && candidate.getNationality().equals("Ukrainian") && periodsInUkr > 10) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
```suggestion
    public boolean test(Candidate candidate) {
        String[] yearLiveInUkraine = candidate.getPeriodsInUkr().split("-");
        int periodsInUkr = Integer.parseInt(yearLiveInUkraine[1])
                - Integer.parseInt(yearLiveInUkraine[0]);
        if (candidate.getAge() >= 35 && candidate.isAllowedToVote()
                && candidate.getNationality().equals("Ukrainian") && periodsInUkr > 10) {
            return true;
        } 
        return false;
    }
```
you must use magic values for 35, 0, 1, 10, "-"